src/common/src/session_config/visibility_mode.rs

```diff
@@ -0,0 +1,111 @@
+// Copyright 2023 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Copyright 2022 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Contains configurations that could be accessed via "set" command.
+
+use std::fmt::Formatter;
+
+use super::{ConfigEntry, CONFIG_KEYS};
+use crate::error::ErrorCode::{self, InvalidConfigValue};
+use crate::error::RwError;
+use crate::session_config::VISIBILITY_MODE;
+
+#[derive(Copy, Default, Debug, Clone, PartialEq, Eq)]
+pub enum VisibilityMode {
+    #[default]
```

Comment:
May I ask the reason to read `Checkpoint` by default? Do we have the plan to change the default value to `All`?

Reply:
+1 to enable VISIBILITY_MODE=all by default since it provides better freshness and normally user doesn't pay serious attention to data rewind.

> shall we even change FLUSH back to always enforce a checkpoint?
This sounds reasonable to me since `FLUSH` normally indicates persistence.

Reply:
> Or shall we even change FLUSH back to always enforce a checkpoint?

+1, agree with Patrick

Reply:
https://github.com/risingwavelabs/risingwave/pull/7188

Reply:
Some earlier discussions: https://github.com/risingwavelabs/risingwave/pull/4966#discussion_r977195601

Should we adopt the `CHECKPOINT` command now?